### PR TITLE
`rm` not submitting everything at once

### PR DESCRIPTION
**Pull request by kkalinowski-reef**
- Ensured that at most `threads` parallel removals are scheduled, so that we don't schedule everything in `ls` at once (there could be millions of files)
- Ensured that with a single thread, `failFast` actually fails as if there were no threads.

### CODE REVIEW

File: b2/console_tool.py [hunk 1]
```diff
@@ -1958,38 +2042,32 @@ def run(self, args):
...
+                    reporter.print_completion(message)
+
+                    failed_on_any_file = True
+                    if args.failFast:
+                        break
+
+                elif event_type == submit_thread.EXCEPTION_TAG:
+                    raise data[0]
+
+            submit_thread.join()
```
Comment (ppolewicz) [inline b2/console_tool.py:2070]: it is possible to get out of this with an exception (`Keyboardinterrupt` for example) where the thead will keep going and the program will be stuck on exiting.

To combat this I propose the thread to be `daemon=True` - see the note about daemon threads [here](https://docs.python.org/3/library/threading.html#thread-objects)

File: b2/console_tool.py [hunk 2]
```diff
@@ -1945,6 +1954,81 @@ class Rm(Ls):
...
+            runner: 'Rm',
+            args: argparse.Namespace,
+            messages_queue: queue.Queue,
+            reporter: ProgressReport,
+        ):
+            self.runner = runner
+            self.args = args
+            self.messages_queue = messages_queue
+            self.reporter = reporter
+            self.semaphore = threading.BoundedSemaphore(value=self.args.threads)
```
Comment (ppolewicz) [inline b2/console_tool.py:1973]: `1*self.args.threads` is not a good value for this - it should be at least `2*self.args.threads` for performance, but generally I'd separate it to a different argparse parameter (as it's a tunable that costs us one line of code)

File: CHANGELOG.md [hunk 3]
```diff
@@ -16,13 +16,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Stop using b2sdk.v1 in arg_parser.py
 * Fix issues when running commands on Python 3.11
 * Fix tests after changes introduced in b2sdk 1.19.0
+* `rm` can handle any number of files
+* Ensured that `ls`/`rm` called `--with_wildcard` and without `--recursive` prints out a proper error, without full stack trace
```
Comment (ppolewicz) [inline CHANGELOG.md:20]:
```suggestion
```
There was no release yet, so we don't have to explain what changed before master and now

File: b2/console_tool.py (hunk 2 above)
Comment (kkalinowski-reef) [reply b2/console_tool.py]: Added `queueSize`, if unset it defaults to `2*threads`.

File: b2/console_tool.py (hunk 1 above)
Comment (kkalinowski-reef) [reply b2/console_tool.py]: Made the thread a daemon.

**Review by ppolewicz — APPROVED** [review]
LGTM, submit to upstream please

**Pull request closed without merging**